File: tests/csharp/tests-csharp-nunit-allure3-selenium/Tests/Ui/HomeErrorStateTests.cs
using Allure.Net.Commons;
using Allure.NUnit.Attributes;
using Api;
using Tests;

namespace Tests.Ui;

[AllureLabel("layer", "ui")]
[AllureEpic("Home")]
[AllureFeature("Error states")]
[AllureSeverity(SeverityLevel.normal)]
[AllureSuite("Home error states (mock)")]
[NonParallelizable]
public sealed class HomeErrorStateTests : TestBase
{
    private bool _mockStandAvailable;

    [SetUp]
    public void RequireMockStand()
    {
        _mockStandAvailable = MockScenarios.Available();
        Assume.That(
            _mockStandAvailable,
            "WireMock admin API is not exposed on this stand — error injection needs the mock profile");
    }

    [TearDown]
    public void ResetScenarios()
    {
        if (_mockStandAvailable)
        {
            MockScenarios.ResetAll();
        }
    }

    [Test]
    [Category("ui")]
    [Category("mock")]
    [Category("negative")]
    [AllureName("Items API failure shows a readable error, not a blank page")]
    public void ItemsApiFailureShowsReadableError()
    {
        MockScenarios.SetState("items", "error");
        HomePage.OpenPage()
            .ShouldShowItemsError("✗ items: HTTP 500");
    }

    [Test]
    [Category("ui")]
    [Category("mock")]
    [Category("negative")]
    [AllureName("Health API failure shows a readable error in the health panel")]
    public void HealthApiFailureShowsReadableError()
    {
        MockScenarios.SetState("health", "error");
        HomePage.OpenPage()
            .ShouldShowHealthError("✗ health: HTTP 500");
    }
}

File: tests/rust/tests-rust-testing-selenium/tests/ui/home_error_state_tests.rs
use allure_cargotest::allure_test;
use tests_rust_testing_selenium as tests;

fn layer() {
    tests::layer_ui("Home error states (mock)", "Home", "Error states", "normal");
    allure_rust_commons::tag("mock");
    allure_rust_commons::tag("negative");
}

#[allure_test(name = "Items API failure shows a readable error, not a blank page")]
#[tokio::test]
async fn items_api_failure_shows_readable_error() {
    layer();
    if !tests::mock_available().await {
        eprintln!(
            "WireMock admin API is not exposed on this stand — error injection needs the mock profile"
        );
        return;
    }
    tests::mock_set_state("items", "error").await;
    tests::with_browser(|| async {
        tests::HomePage::default()
            .open_page()
            .await
            .should_show_items_error("✗ items: HTTP 500")
            .await;
    })
    .await;
    tests::mock_reset_all().await;
}

#[allure_test(name = "Health API failure shows a readable error in the health panel")]
#[tokio::test]
async fn health_api_failure_shows_readable_error() {
    layer();
    if !tests::mock_available().await {
        eprintln!(
            "WireMock admin API is not exposed on this stand — error injection needs the mock profile"
        );
        return;
    }
    tests::mock_set_state("health", "error").await;
    tests::with_browser(|| async {
        tests::HomePage::default()
            .open_page()
            .await
            .should_show_health_error("✗ health: HTTP 500")
            .await;
    })
    .await;
    tests::mock_reset_all().await;
}
